Bottom: d9a72ea37706ceb9a5acf8c104a91bf9b3a69acc
Top:    f095085ed9ab81060970e8b4f9376634e882d75a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-13 01:16:41 +0000

Refresh of control-updates-of-noholes-cac

---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 4d2708a..e975f3b 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -104,6 +104,10 @@ should_compute_no_holes (PolygonType *poly, const BoxType *clip_box)
   float poly_bounding_area;
   float clipped_poly_area;
 
+  /* If there is no passed clip box, compute the whole thing */
+  if (clip_box == NULL)
+    return 1;
+
   x1 = MAX (poly->BoundingBox.X1, clip_box->X1);
   x2 = MIN (poly->BoundingBox.X2, clip_box->X2);
   y1 = MAX (poly->BoundingBox.Y1, clip_box->Y1);
